src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -1649,4 +1649,89 @@ public long countGermplasmByStudy(final Integer studyDbId) {
 		}
 	}
 
+	public List<Germplasm> getExistingCrosses(final String femaleParent, final int methodId, final List<Integer> maleParentIds,
+		final String gid) {
+		try {
+			final StringBuilder builder = this.buildGetExistingCrossesQueryString(maleParentIds, gid);
+			final SQLQuery sqlQuery = this.getSession().createSQLQuery(builder.toString());
+			sqlQuery.setParameterList("maleParentIds", maleParentIds);
+			sqlQuery.setParameter("methodId", methodId);
+			sqlQuery.setParameter("femaleParentId", femaleParent);
+			if(!StringUtils.isEmpty(gid)) {
+				sqlQuery.setParameter("gid", gid);
+			}
+			sqlQuery.addScalar("gid");
+			sqlQuery.addScalar("nval");
+
+			final List<Germplasm> existingCrosses = new ArrayList<>();
+			final List<Object[]> results = sqlQuery.list();
+			for (final Object[] result : results) {
+				final Germplasm existingCross = new Germplasm();
+				existingCross.setGid((Integer) result[0]);
+				existingCross.setGermplasmPeferredName((String) result[1]);
+				existingCrosses.add(existingCross);
+			}
+			return existingCrosses;
+		} catch (final HibernateException e) {
+			final String message = "Error with getExistingCrosses" + e.getMessage();
+			GermplasmDAO.LOG.error(message, e);
+			throw new MiddlewareQueryException(message, e);
+		}
+	}
+
+	StringBuilder buildGetExistingCrossesQueryString(final List<Integer> maleParentIds, final String gid) {
+		final StringBuilder builder = new StringBuilder();
+		builder.append("SELECT g.gid as gid, (select n.nval from names n where n.nstat=1 and n.gid = g.gid limit 1) as nval ");
+		builder.append(" FROM germplsm g ");
+		if(maleParentIds.size() > 1) {
+			builder.append("INNER JOIN progntrs p ON p.gid = g.gid ");
+			builder.append("WHERE g.deleted = 0 AND g.methn = :methodId AND g.gpid1 = :femaleParentId ");
+			if(!StringUtils.isEmpty(gid)) {
+				builder.append("AND g.gid <> :gid ");
+			}
+			builder.append("AND p.pid IN (:maleParentIds) ");
+			builder.append(" AND g.gpid2 IN ( ");
+			builder.append(" SELECT parentId FROM ( ");
+			for (int i = 0; i < maleParentIds.size(); i++) {
```

Comment:
As discussed, find a way to make code more readable. Query performance was tested okay in centos1 > brachiaria